### PR TITLE
documented backend functionalities to the demo

### PR DESCRIPTION
**Pull request by lochoa2**

### CODE REVIEW

**Review by Kelbaso — APPROVED** [review]
Great commenting to help users better understand the demo for the backend. LGTM!

File: src/backend/demo.cpp [hunk 1]
```diff
@@ -1,88 +1,158 @@
...
+    // checks if the item was added correctly
+    if (result != "success") {
+        // same executable error
+        cout << result << endl;
     }
-    // how to add 
-    string result = launcher::add("muck1", "C:\\Program Files (x86)\\Steam\\steamapps\\common\\Muck\\Muck.exe", 
-    "C:\\Program Files (x86)\\Steam\\steam\\games\\goose.ico");
+
+    // adding incorrect executable path
```
Comment (asagrekin) [inline src/backend/demo.cpp:33]: I like the examples of incorrect usage, to show error handling.

File: src/backend/demo.cpp [hunk 2]
```diff
@@ -1,88 +1,158 @@
...
+
+    // delete an executable, calculator
+    string del_result = launcher::del(map["calculator"]);
+    if (del_result != "success") {
+        cout << del_result << endl;
+    }
+
+    // reload the data stored in the database
     games = launcher::get_data();
-    int id;
```
Comment (asagrekin) [inline src/backend/demo.cpp:63]: Good job getting rid of unused code.

**Review by asagrekin — APPROVED** [review]
I see no issues here. Good to merge.

File: src/backend/demo.cpp [hunk 3]
```diff
@@ -1,88 +1,158 @@
...
+        // invalid executable path
         cout << result << endl;
-        return EXIT_FAILURE;
     }
 
-    // string icoResult = launcher::add("game1", "C:\\Program Files (x86)\\Steam\\steamapps\\common\\Muck\\Muck.exe", 
-    // "C:\\Program Files (x86)\\Steam\\steam\\games\\goose.ico");
-    // // checks if the item was added correctly
-    // if (result != "success") {
-    //     // (frontend) check launcher::add for types of errors
```
Comment (slok1368) [inline src/backend/demo.cpp:34]: Nice clean up.

**Review by slok1368 — APPROVED** [review]

**Pull request merged**